cfgmgr/vlanmgr.cpp

```diff
@@ -26,11 +27,26 @@ VlanMgr::VlanMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, c
         m_statePortTable(stateDb, STATE_PORT_TABLE_NAME),
         m_stateLagTable(stateDb, STATE_LAG_TABLE_NAME),
         m_stateVlanTable(stateDb, STATE_VLAN_TABLE_NAME),
+        m_stateVlanMemberTable(stateDb, STATE_VLAN_MEMBER_TABLE_NAME),
         m_appVlanTableProducer(appDb, APP_VLAN_TABLE_NAME),
         m_appVlanMemberTableProducer(appDb, APP_VLAN_MEMBER_TABLE_NAME)
 {
     SWSS_LOG_ENTER();
 
+    if (isWarmStart())
+    {
+        const std::string cmds = std::string("")
+          + IP_CMD + " link show " + DOT1Q_BRIDGE_NAME + " 2>/dev/null";
+
+        std::string res;
+        int ret = swss::exec(cmds, res);
+        if (ret == 0)
+        {
+            // Don't reset vlan aware bridge upon swss docker warm restart.
+            SWSS_LOG_INFO("vlanmgrd warm start, skipping bridge create");
+            return;
```

Comment:
i am not sure if this approach is bullet proof.

what if we later change vlan_filtering option, or enable more option for the bridge. it could happen that older version does not have that option, but new vlanmgrd will enable that option, but the warm reboot will miss it.

I think the right approach should still remove all of them and add new, this is mainly control plane, then we can still achieve non data plane disruption.

Reply:
Removal of vlan doesn't affect data plane directly, but BGP docker and BGP will be affected and cause route flapping.

If there is vlan_filtering option change though unlikely for now,  probably the easier way is to handle that explicitly.

Reply:
I am not particularly worry about vlan_filtering, I am more worry about future bridge attribute, maybe disable unknown multicast, unknown unicast options.

for bgp docker, I think we can do docker pause.

https://docs.docker.com/engine/reference/commandline/pause/

Reply:
Delete/create of bridge and vlan is done in linux kernel,  and zebra listen on that, pausing docker in this case may trigger unknown side effect since we don't know the exact timing of netlink message.   It also makes interface handling more complex.

For disabling unknown multicast, unknown unicast,  probably we should add configuration option for them,  that was in the original vlan trunk pull request.   We may bring it back and refine the change later.

Reply:
my point is that in the future you never know what you are going to add for the bridge. Therefore, we should create exactly the same one as we create in cold boot. 

To ensure that, the cleanest approach is to remove and recreate, then we can share the same code path as the cold boot.

if this can cause control plane disruption, we should then stop the bgp container and do the bgp gr.

---
In reply to: [205924420](https://github.com/Azure/sonic-swss/pull/550#discussion_r205924420) [](ancestors = 205924420)

Reply:
Besides system level warm reboot, we want to support docker warm restart.  We intend to have same code path for cold boot and warm boot whenever possible. 

For this case,  it is in constructor phase.   I don't see why new option has to be put here instead of as a configuration option.   Also we use docker to separate the services,   stopping other docker when doing operation in one docker doesn't seem that clean.